sdk/go/common/util/ciutil/buildkite.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2016-2019, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ciutil
+
+import (
+	"os"
+)
+
+// buildkiteCI represents a Buildkite CI/CD system.
+type buildkiteCI struct {
+	baseCI
+}
+
+// DetectVars detects the env vars for a Buildkite Build.
+func (bci buildkiteCI) DetectVars() Vars {
+	v := Vars{Name: Buildkite}
+	// https://buildkite.com/docs/pipelines/environment-variables#bk-env-vars-buildkite-branch
+	v.BranchName = os.Getenv("BUILDKITE_BRANCH")
+	// https://buildkite.com/docs/pipelines/environment-variables#bk-env-vars-buildkite-build-id
+	v.BuildID = os.Getenv("BUILDKITE_BUILD_ID")
+	// https://buildkite.com/docs/pipelines/environment-variables#bk-env-vars-buildkite-build-number
+	v.BuildNumber = os.Getenv("BUILDKITE_BUILD_NUMBER")
+	// https://buildkite.com/docs/pipelines/environment-variables#bk-env-vars-buildkite-build-url
+	v.BuildURL = os.Getenv("BUILDKITE_BUILD_URL")
+	// https://buildkite.com/docs/pipelines/environment-variables#bk-env-vars-buildkite-message
+	// This is usually the commit message but can be other messages.
+	v.CommitMessage = os.Getenv("BUILDKITE_MESSAGE")
+	// https://buildkite.com/docs/pipelines/environment-variables#bk-env-vars-buildkite-pull-request
+	v.PRNumber = os.Getenv("BUILDKITE_PULL_REQUEST")
```

Comment:
The Buildkite docs state `BUILDKITE_PULL_REQUEST` will be `"false"` for non-PR builds. Can you please add a check here to skip setting the `PRNumber` property if the value is the string `"false"`?

Reply:
Adding this check will also make sure that `PRNumber` is indeed an empty string for non-PR builds which you are checking for at https://github.com/pulumi/pulumi/pull/7933/files#diff-a12ba82451d68131c5f6bb2e2546494b3cbf954c5aac38f7ee575b982134d228R45.

Reply:
I've fixed this and also combined the if statements to make the logic easier.